src/main/java/fi/aalto/cs/apluscourses/model/ModuleSource.java

```diff
@@ -0,0 +1,8 @@
+package fi.aalto.cs.apluscourses.model;
+
+import org.jetbrains.annotations.Nullable;
+
+public interface ModuleSource {
```

Comment:
A short explanatory comment here would be helpful (also since `getModule` isn't documented in `Course`)

Reply:
I'll fix that to not return null. A very good point!